File: src/main/java/com/chattriggers/ctjs/launch/mixins/transformers/ParticleAccessor.java
package com.chattriggers.ctjs.launch.mixins.transformers;

import net.minecraft.client.particle.Particle;
import org.spongepowered.asm.mixin.Mixin;
import org.spongepowered.asm.mixin.gen.Accessor;
import org.spongepowered.asm.mixin.gen.Invoker;

@Mixin(Particle.class)
public interface ParticleAccessor {
    @Accessor
    double getX();

    @Accessor
    double getY();

    @Accessor
    double getZ();

    @Accessor
    double getVelocityX();

    @Accessor
    double getVelocityY();

    @Accessor
    double getVelocityZ();

    @Invoker
    void invokeSetColorAlpha(float alpha);
}

File: src/main/kotlin/com/chattriggers/ctjs/launch/mixins/transformers/ParticleAccessor.kt
package com.chattriggers.ctjs.launch.mixins.transformers

import net.minecraft.client.particle.Particle
import org.spongepowered.asm.mixin.Mixin
import org.spongepowered.asm.mixin.gen.Accessor
import org.spongepowered.asm.mixin.gen.Invoker

@Mixin(Particle::class)
interface ParticleAccessor {
    @Accessor
    fun getX(): Double

    @Accessor
    fun getY(): Double

    @Accessor
    fun getZ(): Double

    @Accessor
    fun getVelocityX(): Double

    @Accessor
    fun getVelocityY(): Double

    @Accessor
    fun getVelocityZ(): Double

    @Invoker
    fun invokeSetColorAlpha(alpha: Float)
}
